exareme2/controller/services/exaflow/controller.py

```diff
@@ -0,0 +1,118 @@
+import asyncio
+from typing import Dict
+from typing import List
+
+from exareme2 import exaflow_algorithm_classes
+from exareme2.controller import logger as ctrl_logger
+from exareme2.controller.federation_info_logs import log_experiment_execution
+from exareme2.controller.services.exaflow.execution_engine import (
+    AlgorithmExecutionEngine,
+)
+from exareme2.controller.services.exaflow.tasks_handler import ExaFlowTasksHandler
+from exareme2.controller.uid_generator import UIDGenerator
+from exareme2.worker_communication import WorkerInfo
+
+
+class WorkerException(Exception):
+    pass
+
+
+class WorkerUnresponsiveException(WorkerException):
+    def __init__(self):
+        super().__init__("One of the workers stopped responding")
+
+
+class WorkerTaskTimeoutException(WorkerException):
+    def __init__(self, timeout):
+        super().__init__(
+            f"Task took longer than {timeout} seconds. Increase timeout or try again."
+        )
+
+
+# Controller class
+class Controller:
+    def __init__(self, worker_landscape_aggregator, task_timeout):
+        self.worker_landscape_aggregator = worker_landscape_aggregator
+        self.task_timeout = task_timeout
+        self.lock = asyncio.Lock()
+
+    def _create_worker_tasks_handler(self, request_id, worker_info: WorkerInfo):
+        worker_addr = f"{worker_info.ip}:{worker_info.port}"
+        return ExaFlowTasksHandler(
+            request_id,
+            worker_id=worker_info.id,
+            worker_queue_addr=worker_addr,
+            tasks_timeout=self.task_timeout,
+        )
+
+    async def exec_algorithm(self, algorithm_name, algorithm_request_dto):
+        async with self.lock:
```

Comment:
We should be able to run 2 concurrent flower algorithms. Will this work? We need to test